### PR TITLE
Add missing implementation in `RoboMenuItem`

### LINKED ISSUE #3085: RoboMenuItem doesn't support setting a title from a resId correctly.
### Description

RoboMenuItem.setTitle(int resId) doesn't set the title - instead, it just returns this. Instead, it feels like it should, well, set the title using the stored context.

### Steps to Reproduce

Write a test that sets a title on a menu item, and observe it is not actually set. :-)

### Robolectric & Android Version

NA - the implementation is clearly visible here: https://github.com/robolectric/robolectric/blob/master/robolectric-shadows/shadows-core/src/main/java/org/robolectric/fakes/RoboMenuItem.java#L79

### PR DESCRIPTION
**Pull request by MGaetan89**
This PR revives #3289 submitted by @jorgegil96 in order to fix #3085.

In addition to the initial fix, this commit also adds:
- Missing implementations on some getters.
- Nullability annotation to match AOSP.
- A couple of tests.

### CODE REVIEW

File: robolectric/src/test/java/org/robolectric/fakes/RoboMenuItemTest.java [hunk 1]
```diff
@@ -54,15 +55,19 @@ public void expandActionView_shouldReturnFalseIfActionViewIsNull() throws Except
 
   @Test
   public void expandActionView_shouldSetExpandedTrue() throws Exception {
-    item.setActionView(new View(ApplicationProvider.getApplicationContext()));
+    View actionView = new View(ApplicationProvider.getApplicationContext());
```
[the remark this answers is not in the document]
Comment (MGaetan89) [reply robolectric/src/test/java/org/robolectric/fakes/RoboMenuItemTest.java]: Done

**Comment by hoisie** [comment]
> > @brettchabot I have a small question: which Nullable/NonNull annotation should we use in Robolectric? javax one or android.annotation one?
> 

I personally prefer `javax.annotation`.

**Comment by MGaetan89** [comment]
I can submit a PR later to migrate every annotation to JavaX later.

@utzcoz we can add a note about in the [contributor guide](https://robolectric.org/contributing/). What do you think?

**Comment by MGaetan89** [comment]
> > @utzcoz we can add a note about in the [contributor guide](https://robolectric.org/contributing/). What do you think?
> 
> LGTM. Should we migrate all annotation to javax one?

I'd say yes, so the rule is applied, and it's easier for contributor to know what to use.

**Pull request merged**